drgn/helpers/linux/cgroup.py

```diff
@@ -0,0 +1,169 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Cgroup
+------
+
+The ``drgn.helpers.linux.cgroup`` module provides helpers for working with
+cgroup interface in :linux:`include/linux/cgroup.h`. Only cgroup v2 is
+supported.
+"""
+
+
+from drgn import NULL, cast, container_of
+from drgn.helpers.linux.kernfs import kernfs_name, kernfs_path
+from drgn.helpers.linux.list import list_for_each_entry
+
+
+__all__ = [
+    'cgroup_name',
+    'cgroup_parent',
+    'cgroup_path',
+    'css_for_each_child',
+    'css_for_each_descendant_pre',
+    'css_next_child',
+    'css_next_descendant_pre',
+    'sock_cgroup_ptr',
+]
+
+
+def sock_cgroup_ptr(skcd):
+    """
+    .. c:function:: struct cgroup *sock_cgroup_ptr(struct sock_cgroup_data *skcd)
+
+    Get cgroup of a socket by ``struct sock_cgroup_data`` member embedded to
+    every ``struct sock``.
+
+
+    :return: ``struct cgroup *`` object.
+    """
+    return cast('struct cgroup *', skcd.val)
+
+
+def cgroup_parent(cgrp):
+    """
+    .. c:function:: struct cgroup *cgroup_parent(struct cgroup *cgrp)
+
+    Get parent of cgroup if it exists.
+
+    :return: ``struct cgroup`` object or None.
+    """
+    parent_css = cgrp.self.parent
+    if parent_css:
+        return container_of(parent_css, 'struct cgroup', 'self')
+    return None
```

Comment:
Ah here's the place I was thinking of when I said I wasn't consistent with this. Let's make this return a `NULL` object, too, then we can get rid of the `:return:` annotation and instead note that it returns `NULL` if the cgroup doesn't have a parent.